client/rpc-servers/src/lib.rs

```diff
@@ -116,14 +119,17 @@ mod inner {
 	/// Start WS server listening on given address.
 	///
 	/// **Note**: Only available if `not(target_os = "unknown")`.
-	pub fn start_ws<M: pubsub::PubSubMetadata + From<jsonrpc_core::futures::sync::mpsc::Sender<String>>> (
+	pub fn start_ws<
+		M: pubsub::PubSubMetadata + From<jsonrpc_core::futures::sync::mpsc::Sender<String>>,
+	>(
 		addr: &std::net::SocketAddr,
 		max_connections: Option<usize>,
 		cors: Option<&Vec<String>>,
 		io: RpcHandler<M>,
+		maybe_max_payload_mb: Option<usize>,
 	) -> io::Result<ws::Server> {
 		ws::ServerBuilder::with_meta_extractor(io, |context: &ws::RequestContext| context.sender().into())
-			.max_payload(MAX_PAYLOAD)
+			.max_payload(maybe_max_payload_mb.map(|mb| mb * MEGABYTE).unwrap_or(DEFAULT_MAX_PAYLOAD))
```

Comment:
Should there be a check against absurd values set here, `u64::MAX` might not work so well? Not sure if this is sanity checked on the `jsonrpc` side?

Reply:
it's an `usize` here and in `jsonrpc` so I think it should be fine

Reply:
We might overflow in this exact line, perhaps better to use `saturating_mul`?